test/util.ts

```diff
@@ -102,13 +102,21 @@ export function transpileAndExecute(
     compilerOptions?: tstl.CompilerOptions,
     luaHeader?: string,
     tsHeader?: string,
+    ambientTsStr = "",
 ): any {
     const wrappedTsString = `${tsHeader ? tsHeader : ""}
         declare function JSONStringify(this: void, p: any): string;
         function __runTest(this: void): any {${tsStr}}`;
 
     const lua = `${luaHeader ? luaHeader : ""}
-        ${transpileString(wrappedTsString, compilerOptions, false)}
+        ${transpileString(
+            {
```

Comment:
Probably better to change the signature of transpileAndExecute  to `tsStr: string | { [filename: string]: string }` (same as what transpileString accepts), and then just pass in this object from your test.